operator-framework/src/test/java/io/javaoperatorsdk/operator/GracefulStopWithConfigurationIT.java

```diff
@@ -0,0 +1,74 @@
+package io.javaoperatorsdk.operator;
+
+import java.time.Duration;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.javaoperatorsdk.operator.junit.LocallyRunOperatorExtension;
+import io.javaoperatorsdk.operator.sample.gracefulstop.GracefulStopTestCustomResource;
+import io.javaoperatorsdk.operator.sample.gracefulstop.GracefulStopTestCustomResourceSpec;
+import io.javaoperatorsdk.operator.sample.gracefulstop.GracefulStopTestReconciler;
+
+import static io.javaoperatorsdk.operator.sample.gracefulstop.GracefulStopTestReconciler.*;
+import static org.assertj.core.api.Assertions.*;
+import static org.awaitility.Awaitility.*;
+
+public class GracefulStopWithConfigurationIT {
+
+  public static final String TEST_1 = "test1";
+
+  @RegisterExtension
+  LocallyRunOperatorExtension operator =
+      LocallyRunOperatorExtension.builder()
+          .withConfigurationService(o -> o.withCloseClientOnStop(false)
+              .withReconciliationTerminationTimeout(Duration.ofSeconds(RECONCILER_SLEEP)))
```

Comment:
`.withReconciliationTerminationTimeout(Duration.ofSeconds(RECONCILER_SLEEP)))`

User can configure the graceful shutdown time in the Operator settings.